app/actions/websocket/group.test.ts

```diff
@@ -0,0 +1,280 @@
+                    }),
+                },
+            } as WebSocketMessage;
+
+            await handleGroupTeamAssociatedEvent(serverUrl, msg);
+
+            const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            expect(operator.handleGroupTeamsForTeam).toHaveBeenCalledWith({
+                teamId,
+                groups: [{id: groupId}],
+                prepareRecordsOnly: false,
+            });
+        });
+    });
+
+    describe('handleGroupTeamDissociateEvent', () => {
+        it('should handle group team dissociation successfully', async () => {
+            const msg = {
+                data: {
+                    group_team: JSON.stringify({
+                        group_id: groupId,
+                        team_id: teamId,
+                    }),
+                },
+            } as WebSocketMessage;
+
+            jest.mocked(generateGroupAssociationId).mockReturnValue('association-id');
+
+            await handleGroupTeamDissociateEvent(serverUrl, msg);
+
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            expect(deleteGroupTeamById).toHaveBeenCalledWith(database, 'association-id');
+        });
+    });
+
+    describe('handleGroupChannelAssociatedEvent', () => {
+        it('should handle group channel association successfully', async () => {
+            const msg = {
+                data: {
+                    group_channel: JSON.stringify({
+                        group_id: groupId,
+                        channel_id: channelId,
+                    }),
+                },
+            } as WebSocketMessage;
+
+            await handleGroupChannelAssociatedEvent(serverUrl, msg);
+
+            const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            expect(operator.handleGroupChannelsForChannel).toHaveBeenCalledWith({
+                channelId,
+                groups: [{id: groupId}],
+                prepareRecordsOnly: false,
+            });
+        });
+    });
+
+    describe('handleGroupChannelDissociateEvent', () => {
+        it('should handle group channel dissociation successfully', async () => {
+            const msg = {
+                data: {
+                    group_channel: JSON.stringify({
+                        group_id: groupId,
+                        channel_id: channelId,
+                    }),
+                },
+            } as WebSocketMessage;
+
+            jest.mocked(generateGroupAssociationId).mockReturnValue('association-id');
+
+            await handleGroupChannelDissociateEvent(serverUrl, msg);
+
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            expect(deleteGroupChannelById).toHaveBeenCalledWith(database, 'association-id');
+        });
+    });
+
+    describe('error handling', () => {
+        it('should handle errors and fetch groups on failure', async () => {
```

Comment:
I'm not sure which errors are being handled here

Reply:
This tests the branches of the `handleError` function